File: app/src/main/java/com/angogasapps/myfamily/database/MessageDao.java
package com.angogasapps.myfamily.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.angogasapps.myfamily.models.Message;

import java.util.List;

@Dao
public interface MessageDao {
    @Query("SELECT * FROM Message")
    List<Message> getAll();

    @Query("SELECT * FROM Message WHERE id = :id")
    Message getById(String id);

//    @Insert
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Message message);

    @Update
    void update(Message message);

    @Delete
    void delete(Message message);
}

File: app/src/main/java/com/angogasapps/myfamily/database/MessageDao.kt
package com.angogasapps.myfamily.database

import androidx.room.*
import com.angogasapps.myfamily.models.Message

@Dao
interface MessageDao {
    @get:Query("SELECT * FROM Message")
    val all: List<Message>

    @Query("SELECT * FROM Message WHERE id = :id")
    fun getById(id: String): Message

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(message: Message)

    @Update
    fun update(message: Message)

    @Delete
    fun delete(message: Message)
}